src/Spectre.Angular2Client/src/app/heatmap/heatmap.service.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { TestBed, inject } from '@angular/core/testing';
+import { Http, BaseRequestOptions, Response, ResponseOptions } from '@angular/http';
+import { MockBackend, MockConnection } from '@angular/http/testing';
+
+import { HeatmapService } from './heatmap.service';
+
+describe('HeatmapService', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+          MockBackend,
+          BaseRequestOptions,
+          {
+            provide: Http,
+            useFactory: (backendInstance: MockBackend, defaultOptions: BaseRequestOptions) => {
+              return new Http(backendInstance, defaultOptions);
+            },
+            deps: [MockBackend, BaseRequestOptions]
+          },
+          HeatmapService
+      ]
+    });
+  });
+
+  it('should inject service', inject([HeatmapService], (service: HeatmapService) => {
```

Comment:
More tests are necessary - you should fake REST API response and check, whether proper values are set in the output structures. Basically, that's why we are mocking `Http` here.